src/joystick/usb_ids.h

```diff
@@ -67,6 +68,7 @@
 #define USB_PRODUCT_GOOGLE_STADIA_CONTROLLER              0x9400
 #define USB_PRODUCT_EVORETRO_GAMECUBE_ADAPTER1            0x1843
 #define USB_PRODUCT_EVORETRO_GAMECUBE_ADAPTER2            0x1846
+#define USB_PRODUCT_FLYDIGI_VADER4_PRO                    0x2412
```

Comment:
FYI, this product ID is used by every FlyDigi controller, but they all have different report structures. We need a way to positively identify that this is the Vader 4 Pro controller.

Reply:
Interesting. I tried this out with the Vader 2, and it has a 32-byte report and rumble works, but gyro data doesn't come through.

Reply:
Actually, the gyro data is in bytes 4-6 of the packet, for the Vader 2. It looks like it doesn't have an accelerometer sensor.

Reply:
This driver also works for the Apex 3. If the report format is stable across all FlyDigi controllers, then we can accept this patch, thanks!

What work do you have left before merging? Don't worry about the macOS build failure, I can fix that once this is merged.

Reply:
There are two paths for parsing the payload in the Space Station software: for "K1" models and the rest, from what I see it should be just for Apex 3.
There are some operations with "masks" I did not closely look into yet that are different for the two paths, will check later.

Reply:
I'm guessing the get config report will contain the gyro and accelerometer calibration data. That might give us a single path to get the accelerometer scale that works with all controllers.

Reply:
These configs look more like profiles with remappings, stick curves etc, the only motion-related stuff there is whether to emulate mouse or right stick and corresponding sensitivity.
Dualsense emulation seems to be based on vigem, maybe it has some hardcoded calibration data?